core/src/main/java/com/scalar/db/storage/cassandra/CassandraAdmin.java

```diff
@@ -1,69 +1,279 @@
+    } catch (RuntimeException e) {
+      throw new ExecutionException(
+          String.format(
+              "dropping the %s table failed", getFullTableName(keyspacePrefix, namespace, table)),
+          e);
+    }
+    metadataManager.deleteTableMetadata(namespace, table);
+    if (metadataManager.getTableNames(namespace).isEmpty()) {
+      String dropKeyspace =
+          SchemaBuilder.dropKeyspace(fullKeyspace(namespace)).ifExists().getQueryString();
+      try {
+        clusterManager.getSession().execute(dropKeyspace);
+      } catch (RuntimeException e) {
+        throw new ExecutionException(
+            String.format("dropping the %s keyspace failed", fullKeyspace(namespace)), e);
+      }
+    }
   }
 
   @Override
   public void truncateTable(String namespace, String table) throws ExecutionException {
-    throw new UnsupportedOperationException("implement later");
+    String truncateTableQuery =
+        QueryBuilder.truncate(fullKeyspace(namespace), table).getQueryString();
+    try {
+      clusterManager.getSession().execute(truncateTableQuery);
+    } catch (RuntimeException e) {
+      throw new ExecutionException(
+          String.format(
+              "truncating the %s table failed", getFullTableName(keyspacePrefix, namespace, table)),
+          e);
+    }
   }
 
   @Override
   public TableMetadata getTableMetadata(String namespace, String table) throws ExecutionException {
     try {
-      return metadataManager.getTableMetadata(fullNamespace(namespace), table);
+      return metadataManager.getTableMetadata(namespace, table);
     } catch (StorageRuntimeException e) {
       throw new ExecutionException("getting a table metadata failed", e);
     }
   }
 
-  private String fullNamespace(String namespace) {
-    return namespacePrefix.map(s -> s + namespace).orElse(namespace);
+  private String fullKeyspace(String keyspace) {
+    return getFullNamespaceName(keyspacePrefix, keyspace);
+  }
+
+  @VisibleForTesting
+  void createKeyspaceIfNotExists(String fullKeyspace, Map<String, String> options)
+      throws ExecutionException {
+    CreateKeyspace query = SchemaBuilder.createKeyspace(fullKeyspace).ifNotExists();
+    String replicationFactor = options.getOrDefault(REPLICATION_FACTOR, "1");
+    NetworkStrategy networkStrategy =
+        options.containsKey(NETWORK_STRATEGY)
+            ? NetworkStrategy.fromString(options.get(NETWORK_STRATEGY))
+            : NetworkStrategy.SIMPLE_STRATEGY;
+    Map<String, Object> replicationOptions = new LinkedHashMap<>();
+    if (networkStrategy == NetworkStrategy.SIMPLE_STRATEGY) {
+      replicationOptions.put("class", NetworkStrategy.SIMPLE_STRATEGY.toString());
+      replicationOptions.put("replication_factor", replicationFactor);
+    } else if (networkStrategy == NetworkStrategy.NETWORK_TOPOLOGY_STRATEGY) {
+      replicationOptions.put("class", NetworkStrategy.NETWORK_TOPOLOGY_STRATEGY.toString());
+      replicationOptions.put("dc1_name", replicationFactor);
```

Comment:
```suggestion
      replicationOptions.put("dc1", replicationFactor);
```
I think `cassandra-rackdc.properties` uses `dc1` by default, but any particular reason for using `dc1_name`?

Reply:
Indeed, thank you. I copied the `dc1_name` from the Cassandra documentation page but it was likely just a placeholder name for the real datacenter name